src/flask_jsonrpc/site.py

```diff
@@ -173,9 +175,17 @@ def dispatch(
         self, req_json: t.Dict[str, t.Any]
     ) -> t.Tuple[t.Any, int, t.Union[Headers, t.Dict[str, str], t.Tuple[str], t.List[t.Tuple[str]]]]:
         params = req_json.get('params', {})
-        view_func = self.view_funcs.get(req_json['method'])
+        method_name = req_json['method']
+        view_func = self.view_funcs.get(method_name)
+        validate = getattr(view_func, 'jsonrpc_validate', settings.DEFAULT_JSONRPC_METHOD['VALIDATE'])
+        notification = getattr(view_func, 'jsonrpc_notification', settings.DEFAULT_JSONRPC_METHOD['NOTIFICATION'])
         if not view_func:
-            raise MethodNotFoundError(data={'message': f"Method not found: {req_json['method']}"})
+            raise MethodNotFoundError(data={'message': f"Method not found: {method_name}"})
+
+        if self.is_notification_request(req_json) and not notification:
+            raise InvalidRequestError(
+                data={'message': f"The method '{method_name}' doesn't allow Notification Request object"}
```

Comment:
Maybe worth noting what's that Notification Request is, for uninitiated :) .

For example:
```python
f"The method '{method_name}' doesn't allow Notification Request object (without 'id' field)"
```

or

```python
f"The method '{method_name}' doesn't allow Notification Request object (absent 'id')"
```

Reply:
The terminology from JSON-RPC spec is: A Notification is a Request object without an "id" member. 

```
"The method '{method_name}' doesn't allow Notification Request object (without an 'id' member)"
```

What do you think?

Reply:
The more standard the better :)